File: result/metagpt/claude/humaneval/19.c
#include <stdio.h>
#include <string.h>
#include <stdlib.h>

char* sort_numbers(const char* numbers) {
    // 定义字符串到数字的映射
    const char* tonum_keys[] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    int tonum_values[] = {0, 1, 2, 3, 4, 5, 6, 7, 8, 9};
    
    // 定义数字到字符串的映射
    const char* numto_keys[] = {"zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"};
    
    // 初始化计数数组
    int count[10] = {0};
    
    // 复制输入字符串以便处理
    char* input = strdup(numbers);
    char* token = strtok(input, " ");
    
    // 统计每个数字出现的次数
    while (token != NULL) {
        for (int i = 0; i < 10; i++) {
            if (strcmp(token, tonum_keys[i]) == 0) {
                count[i]++;
                break;
            }
        }
        token = strtok(NULL, " ");
    }
    
    // 构建输出字符串
    char* out = (char*)malloc(1000 * sizeof(char));
    out[0] = '\0';
    for (int i = 0; i < 10; i++) {
        for (int j = 0; j < count[i]; j++) {
            strcat(out, numto_keys[i]);
            strcat(out, " ");
        }
    }
    
    // 去除最后一个空格
    if (strlen(out) > 0) {
        out[strlen(out) - 1] = '\0';
    }
    
    // 释放内存
    free(input);
    
    return out;
}

File: result/metagpt/gpt4o/humaneval/19.rs
fn sort_numbers(numbers: &str) -> String {
    // Define the mapping from string to number
    let tonum_keys = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"];
    let mut count = [0; 10];

    // Split the input string and count occurrences of each number
    for token in numbers.split_whitespace() {
        if let Some(pos) = tonum_keys.iter().position(|&key| key == token) {
            count[pos] += 1;
        }
    }

    // Build the output string
    let mut out = String::new();
    for (i, &num) in count.iter().enumerate() {
        for _ in 0..num {
            if !out.is_empty() {
                out.push(' ');
            }
            out.push_str(tonum_keys[i]);
        }
    }

    out
}

fn main() {
    let numbers = "three one four one five nine";
    let sorted_numbers = sort_numbers(numbers);
    println!("Sorted numbers: {}", sorted_numbers);
}